themes/default/content/docs/concepts/inputs-outputs/all.md

```diff
@@ -165,7 +165,193 @@ Server=tcp:myDbServer.database.windows.net;initial catalog=myExampleDatabase;
 
 There is an easier way to generate a concatenated string value using multiple outputs, and that is by using interpolation. Pulumi exposes interpolation helpers that enables you to create strings that contain outputs. These interpolation methods wrap [all](/docs/concepts/inputs-outputs/all/) and [apply](/docs/concepts/inputs-outputs/apply/) with an interface that resembles your language's native string formatting functions. The below example demonstrates how to create a URL from the hostname and port output values of a web server.
 
-{{< example-program path="aws-simulated-server-interpolate" >}}
+{{< chooser language "javascript,typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+const pulumi = require("@pulumi/pulumi");
+/* Simulated outputs of a webServer resource
```

Comment:
I know you're trying to get this wrapped, so apologies for slowing it down! 😇 But I do think this example and the others on this page would be better served if we used actual resources and outputs. That'd make this both clearer and more testable, IMO.

The simulated outputs _look_ simpler on the surface -- but in a way they're actually less clear in that you have to imagine what they stand in for. It'd be better if we just used a real resource or two. We could use just about anything, but here's an example that uses a bucket and bucket object to construct the two URLs the current example exports at the end:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const bucket = new aws.s3.Bucket("bucket");

const file = new aws.s3.BucketObject("bucket-object", {
    bucket: bucket.id,
    key: "some-file.txt",
    content: "some-content",
});

// concat takes a list of args and concatenates all of them into a single output:
export const s3Url1: pulumi.Output<string> = pulumi.concat("s3://", bucket.bucket, "/", file.key);

// interpolate takes a JavaScript template literal and expands outputs correctly:
export const s3Url2: pulumi.Output<string> = pulumi.interpolate`s3://${bucket.bucket}/${file.key}`;
```

Any reason _not_ to do this? I just think it'd be better all 'round. What do you think?

Reply:
Hi @cnunciato ,

Please see my comment in [this thread](https://pulumi.slack.com/archives/C026M35HVDJ/p1709534734659279?thread_ts=1709074658.591829&cid=C026M35HVDJ). Until I can get example code across all languages that accurately reflects what the experience of interacting with a resource is like, the example output is the best I can do in the interim. I don't think this should be something that blocks as we can always add the interactive code later.

Reply:
I was kind of hoping to avoid reinventing the wheel on examples where possible, especially given that it can be challenging recreating the same functionality across the different languages in a way that the code actually works. I will work on updating with the above but it will definitely extend my timeline on wrapping this up.